docs/docs/spaces.mdx

```diff
@@ -0,0 +1,271 @@
+That is because in a regular rasa assistant all intents, entities, and actions are
+available at any time, so every new capability has to play nicely with all existing
+ones. Spaces introduce a basic separation between parts of the bot based on activation
+and deactivation of groups of [rasa primitives](./glossary.mdx#rasa-primitive).
+Every of these groups is called a space.
+Spaces are activated when certain designated intents, also called entry intents, have
+been predicted. With the space activation, all the primitives in the space become
+available for subsequent interactions with the user. Good ways to think about Spaces are
+
+* that they allow you to specify follow-up primitives, which only become accessible after another intent has come beforehand
+* that they are like multiple sub-bots merged into one with the possibility of sharing primitives
+
+
+## When to use Spaces
+
+Spaces can be helpful when you are dealing with multiple domains of your business in
+a single assistant. Oftentimes, this leads to a multitude of forms, entities, and inform
+intents that start to overlap at some point. Form filling and inform intents are a
+typical case of having this follow-up structure mentioned above. Another good case is
+when you want to be able to define different behavior for help- or clarification
+requests based on the subdomain or process the user is in. This is technically possible
+today with stories, but it can be cumbersome to describe the full event horizon
+for every interaction route.
+
+## When not to use Spaces / Limitations
+
+Because the Spaces feature splits a rasa bot into multiple parts, it is interacting with almost
+every of the existing features of rasa. We made it work with almost all of them, but
+there are some exceptions:
+
+* if you have created extensive customization and introduced your own data hierarchies
+or post-processing steps, these might not be compatible with Spaces.
+* Spaces currently do not support stories. This is a big limitations for many
+existing bots. However, because stories and their existing policies work with a fixed
+event horizon (`max_history`), they are at this point not compatible with the idea of
+being able to describe encapsulated units of logic well.
+* Entity extraction is currently limited to an adapted version of
+the `CRFEntityExtractor` which works with Spaces boundaries. You can find more on this
+in the following sections.
+
+
+## An example Spaces bot
+We have created a bot using Spaces for you to look at, learn from and experiment with.
+It features three different spaces in the financial domain. [Here is the repository](https://github.com/RasaHQ/financial-spaces-bot).
+
+
+## How to use Spaces
+
+Spaces are defined by separate sets of domain, nlu, and rule files. These separate
+sets are then assembled to create a unified assistant. The assembly plan needs to be
+added to the `config.yml` using the newly introduces `spaces` key. Second, you need
```

Comment:
Shouldn't this be in `domain.yml` since it's focused on the training data primitives that are also in `domain.yml` and not the pipeline and polices?

Reply:
✅ replied on Slack [here](https://rasa-hq.slack.com/archives/C01TD639GMQ/p1684480173880869?thread_ts=1684337607.585739&cid=C01TD639GMQ) and [here](https://rasa-hq.slack.com/archives/C01TD639GMQ/p1684482353729519?thread_ts=1684337607.585739&cid=C01TD639GMQ). This is something we can revisit with customer feedback for the GA